sources/packages/api/src/route-controllers/institution/institution.controller.ts

```diff
@@ -446,4 +452,84 @@ export class InstitutionController extends BaseController {
       },
     }));
   }
+
+  /**
+   * Get the Institution details for the ministry institution detail page
+   * @param institutionId
+   * @returns AESTInstitutionDetailDto
+   */
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Groups(UserGroups.AESTUser)
+  @Get("/:institutionId/detail")
+  async getAESTInstitutionDetailById(
+    @Param("institutionId") institutionId: number,
+  ): Promise<AESTInstitutionDetailDto> {
+    if (!institutionId) {
+      throw new UnprocessableEntityException(
+        "Institution Id is required to fetch the institution detail",
+      );
+    }
+    const institutionDetail =
+      await this.institutionService.getAESTInstitutionDetailById(institutionId);
+    return {
+      legalOperatingName: institutionDetail.legalOperatingName,
+      operatingName: institutionDetail.operatingName,
+      primaryPhone: institutionDetail.primaryPhone,
+      primaryEmail: institutionDetail.primaryEmail,
+      website: institutionDetail.website,
+      regulatingBody: institutionDetail.regulatingBody,
+      institutionTypeName: institutionDetail.institutionType.name,
+      formattedEstablishedDate: dayjs(institutionDetail.establishedDate).format(
+        EXTENDED_DATE_FORMAT,
+      ),
+      primaryContactEmail:
+        institutionDetail.institutionPrimaryContact.primaryContactEmail,
+      primaryContactFirstName:
+        institutionDetail.institutionPrimaryContact.primaryContactFirstName,
+      primaryContactLastName:
+        institutionDetail.institutionPrimaryContact.primaryContactLastName,
+      primaryContactPhone:
+        institutionDetail.institutionPrimaryContact.primaryContactPhone,
+      legalAuthorityEmail:
+        institutionDetail.legalAuthorityContact.legalAuthorityEmail,
+      legalAuthorityFirstName:
+        institutionDetail.legalAuthorityContact.legalAuthorityFirstName,
+      legalAuthorityLastName:
+        institutionDetail.legalAuthorityContact.legalAuthorityLastName,
+      legalAuthorityPhone:
+        institutionDetail.legalAuthorityContact.legalAuthorityPhone,
+      address: {
+        addressLine1: institutionDetail.institutionAddress.addressLine1,
+        addressLine2: institutionDetail.institutionAddress.addressLine2,
+        city: institutionDetail.institutionAddress.city,
+        country: institutionDetail.institutionAddress.country,
+        provinceState: institutionDetail.institutionAddress.provinceState,
+        postalCode: institutionDetail.institutionAddress.postalCode,
+      },
+    };
+  }
+
+  /**
+   * Get the Basic Institution info for the ministry institution detail page
+   * @param institutionId
+   * @returns AESTInstitutionDetailDto
+   */
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Groups(UserGroups.AESTUser)
+  @Get("/:institutionId")
+  async getBasicInstitutionInfoById(
+    @Param("institutionId") institutionId: number,
+  ): Promise<BasicInstitutionInfo> {
+    if (!institutionId) {
```

Comment:
I believe that if the id is not provided the route is not going to be satisfied hence this controller will not even be called.
Can you please check? I believe that this validation is not necessary.